### DISCUSSION TITLE
What is the best way to heavily customized landing page?

### DISCUSSION THREAD

**Discussion by cemrehancavdar** [squidfunk/mkdocs-material#5426] [Q&A] [answered]
I have been developing a custom landing page that features a unique design and incorporates custom JavaScript codes to enhance interactivity. The UI is separate from the documentation, but it includes a summary of how to use and install the product. At the bottom of the page, there will be a section featuring code and preview capabilities, allowing users to see changes in real-time.

I am wondering about the best way to implement this. Should I extend the base.html file and override its blocks? Or should I create custom patterns, styles, and JavaScript files? Is there an obvious approach that I should take?

**Comment by squidfunk** [comment, marked as answer]
I recommend looking [at our overrides](https://github.com/squidfunk/mkdocs-material/tree/master/src/.overrides) – the best way is to create a separate template called `home.html` and then override blocks selectively. You should never override `base.html` directly, only extend it, as that will make upgrading challenging.

**Comment by cemrehancavdar** [comment]
Given my scenario, it seems more reasonable to create a simple landing page that references the URL to the documentation, rather than modifying the main.html file.

**Reply by squidfunk** [reply]
What ever works best for you ☺️
